scipy/_lib/meson.build

```diff
@@ -190,6 +190,7 @@ py3.install_sources(
     'array_api_compat/array_api_compat/numpy/__init__.py',
     'array_api_compat/array_api_compat/numpy/_aliases.py',
     'array_api_compat/array_api_compat/numpy/_typing.py',
+    'array_api_compat/array_api_compat/numpy/_info.py',
```

Comment:
I needed this locally as well (and the CI fail seems to be needing the `torch` one too):

```diff
--- a/scipy/_lib/meson.build
+++ b/scipy/_lib/meson.build
@@ -163,6 +163,7 @@ py3.install_sources(
     'array_api_compat/array_api_compat/cupy/__init__.py',
     'array_api_compat/array_api_compat/cupy/_aliases.py',
     'array_api_compat/array_api_compat/cupy/_typing.py',
+    'array_api_compat/array_api_compat/cupy/_info.py',
     'array_api_compat/array_api_compat/cupy/fft.py',
     'array_api_compat/array_api_compat/cupy/linalg.py',
   ],
```

if I understand your comment above correctly, I believe this is a best practice in both `meson` and CMake for faster build experience. I do know some teams that use glob patterns instead of maintaining lists, but best practice in both those build systems has mostly been to maintain the lists IIRC.

I believe there is some other options via subprojects or whatever each of them calls it when they provide the appropriate build system-native infra themselves.

Reply:
(so if `array_api_compat` maintained its own meson build system/list like above I think there'd be a way to avoid it perhaps, but Ralf would know better than I; might be dumping too much on Aaron for the benefit though)

Reply:
Tyler is right - instant rebuilds and not installing unnecessary files have a price, and that price is "no globbing". See this FAQ entry: https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard

> (so if `array_api_compat` maintained its own meson build system/list like above I think there'd be a way to avoid it perhaps, but Ralf would know better than I; might be dumping too much on Aaron for the benefit though)

Indeed, it's too much work upstream, and Meson isn't the build system I would choose for a small pure Python project probably. Maintaining the file listing here is easier overall.